lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -5527,9 +5527,22 @@ void swift::diagnoseConformanceFailure(Type T,
 
       auto rawType = enumDecl->getRawType();
 
-      diags.diagnose(enumDecl->getInherited()[0].getSourceRange().Start,
-                     diag::enum_raw_type_nonconforming_and_nonsynthable,
-                     T, rawType);
+      // Existential types are not literal convertible.
+      if (rawType->is<ExistentialType>()) {
+        SourceRange sourceRange =
```

Comment:
This `sourceRange` could be hoisted out of the if since the value of `enumDecl->getInherited()[0].getSourceRange()` is used on both paths.